app/lib/add_service.rb

```diff
@@ -29,8 +29,7 @@ def initialize(service_request, service_id, requester, new_request=false, confir
 
   def generate_new_service_request
     @service_request.save
-    new_line_items = @service_request.create_line_items_for_service(service: @service, optional: true, recursive_call: false ) || []
-    create_sub_service_requests(new_line_items)
+    @service_request.create_line_items_for_service(service: @service, requester: @requester) || []
```

Comment:
Everything is moved to models - `models/service_request.rb` and `models/sub_service_request.rb`